src/android/Geolocation.java

```diff
@@ -259,48 +388,134 @@ public void onLocationResultError(LocationContext locationContext, LocationError
         locationContext.getCallbackContext().sendPluginResult(result);
     }
 
-    private void requestLocationUpdatesIfSettingsSatisfied(final LocationContext locationContext, final LocationRequest request) {
+    @Override
+    public void onLocationResultSuccessHuawei(LocationContext locationContext, com.huawei.hms.location.LocationResult locationResult) {
+        for (Location location : locationResult.getLocations()) {
+            try {
```

Comment:
The code inside the `try` and the `catch` statements has lots of similarities (sometimes is equally the same). Maybe we could refactor this to avoid code replication?